### PR TITLE
Version control integration documentation reworked

### PR DESCRIPTION
**Pull request by comradekingu**

### CODE REVIEW

File: docs/vcs.rst [hunk 1]
```diff
@@ -34,180 +34,173 @@ SSH repositories
...
+environment variables, (as described in the [cURL documentation](https://curl.haxx.se/docs/))
+or by enforcing it in the VCS configuration, for example:
 
 .. code-block:: sh
 
     git config --global http.proxy http://user:password@proxy.example.com:80
 
 .. note::
 
-    The proxy setting needs to be done in the same context which is used to
```
Comment (comradekingu) [inline docs/vcs.rst]: > The proxy setting needs to be done in the same context used to run Weblate. For the environment it should be set for both WSGI and Celery servers. The VCS configuration has to be set for the user running Weblate.

Could this be explained?

File: docs/vcs.rst [hunk 2]
```diff
@@ -34,180 +34,173 @@ SSH repositories
...
+to access your repo (for example ``git@github.com:WeblateOrg/weblate.git```).
+.
 
 Customizing Git configuration
 +++++++++++++++++++++++++++++
 
-Weblate invokes all VCS commands with HOME pointed to ``home`` directory in
-:setting:`DATA_DIR`, therefore if you want to edit user configuration, you need
-to do this in ``DATA_DIR/home/.git``.
+Weblate invokes all VCS commands with `$HOME` pointed to he ``home`` directory in
```
Comment (comradekingu) [inline docs/vcs.rst]: Symlinked? Hoping `$HOME` is correct.

File: docs/vcs.rst [hunk 3]
```diff
@@ -228,52 +220,50 @@ GitHub
 
 .. versionadded:: 2.3
 
-This just adds a thin layer on top of :ref:`vcs-git` to allow push translation
-changes as pull requests instead of pushing directory to the repository.
-It currently uses the `hub`_ tool to do the integration.
+This adds a thin layer atop :ref:`vcs-git` using the `hub`_ tool to allow pushing
+translation changes as pull requests, instead of
+pushing the directory to the repository.
```
Comment (comradekingu) [inline docs/vcs.rst]: the whole
```suggestion
pushing the whole directory to the repository.
```

File: docs/vcs.rst [hunk 4]
```diff
@@ -228,52 +220,50 @@ GitHub
...
 
 .. _github-push:
 
-Pushing changes to GitHub as pull request
+Pushing changes to GitHub as pull requests
 +++++++++++++++++++++++++++++++++++++++++
 
-If you are translating a project that's hosted on GitHub and don't want to
-push translations to the repository, you can have them sent as a pull request instead.
+If not wanting to push translations to a GitHub repository, they can be sent as either
```
Comment (comradekingu) [inline docs/vcs.rst]: Maybe the squashing add-on should be explained or linked to?
(Same for GitLab)

File: docs/vcs.rst [hunk 5]
```diff
@@ -228,52 +220,50 @@ GitHub
...
 :ref:`github-push` requires a configured `hub`_ installation on your server.
-Follow the installation instructions at https://hub.github.com/ and perform an
-action with `hub`_ to finish the configuration, for example:
+Follow the installation instructions at https://hub.github.com/ use `hub`_ to
+finish the configuration, for example:
 
 .. code-block:: sh
 
-    # DATA_DIR is set in Weblate settings.py, set it accordingy.
+    # DATA_DIR is configured in Weblate settings.py, set it accordingy.
```
Comment (comradekingu) [inline docs/vcs.rst]: According to?

File: docs/vcs.rst [hunk 6]
```diff
@@ -363,14 +352,14 @@ Local files
...
+imported by uploading them. Later you can replace individual files by file upload,
+or add translation strings directly from Weblate (currently available only for
 monolingual translations).
 
-In the background Weblate creates Git repository for you and all changes are
-tracked in in. In case you decide later to use VCS to store the translations,
-it's already within Weblate and you can base on that.
+In the background Weblate creates a Git repository for you and all changes are
+tracked in in. In case you later decide to use a VCS to store the translations,
+you already have a repo within Weblate can base your integration on.
```
Comment (comradekingu) [inline docs/vcs.rst]: Not 100% sure this is the correct way of saying it.

File: docs/vcs.rst (hunk 3 above)
Comment (nijel) [reply docs/vcs.rst]: It should actually be "pushing directly to the repository".

File: docs/vcs.rst (hunk 2 above)
Comment (nijel) [reply docs/vcs.rst]: Yes, I've further improved it.

File: docs/vcs.rst (hunk 1 above)
Comment (nijel) [reply docs/vcs.rst]: Improved.

File: docs/vcs.rst [hunk 7]
```diff
@@ -228,52 +220,50 @@ GitHub
...
+Configure the `hub`_ command line tool and set :setting:`GITHUB_USERNAME` for this to work.
 
 .. seealso::
 
    :setting:`GITHUB_USERNAME`, :ref:`hub-setup` for configuration instructions
 
 .. _hub-setup:
 
-Setting up hub
+Setting up Hub
```
Comment (nijel) [inline docs/vcs.rst]: The tool seems to be called "hub", see https://hub.github.com/

File: docs/vcs.rst (hunk 5 above)
Comment (nijel) [reply docs/vcs.rst]: To match what is specified in the settings.py

File: docs/vcs.rst [hunk 8]
```diff
@@ -390,48 +379,47 @@ while :ref:`vcs-gitlab` creates merge request.
...
-    # Is is /app/data in Docker
+    # It is /app/data in Docker
     $ HOME=${DATA_DIR}/home lab
-    Enter GitLab host (default: https://gitlab.com):
+    Enter the GitLab host (default: https://gitlab.com):
     Create a token here: https://gitlab.com/profile/personal_access_tokens
-    Enter default GitLab token (scope: api):
-    Config saved to ~/.config/lab.hcl
+    Enter the default GitLab token (scope: api):
+    (The config is saved to ~/.config/lab.hcl)
```
Comment (nijel) [inline docs/vcs.rst]: This is output of the lab program, in case you want to improve it, please submit it to them :-).

**Pull request merged**